marketpulse/main/static/js/price_new.js

```diff
@@ -24,15 +24,20 @@ $(document).ready(function() {
     }
 
     $(document).on('click', '.hasplan-check', function () {
-        var n = $(this.checked).length;
+        var choice = $(this).closest('.switch').nextAll('.hasplan').slice(0, 4);
 
-        if (n === 0) {
-            $(this).closest('.planprices').find('.hasplan').hide();
+        if (this.checked) {
+            choice.show();
         } else {
-            $(this).closest('.planprices').find('.hasplan').show();
+            choice.hide();
         }
```

Comment:
Maybe you could also amend the `displayLocationForm` function a few lines above, to use the same style of code as this. It seems more clear the way you implemented it.